app/policies/placements/user_membership_policy.rb

```diff
@@ -0,0 +1,13 @@
+class Placements::UserMembershipPolicy < ApplicationPolicy
+  def destroy?
+    return false unless user != record.user
+
+    user.support_user? ||
+      user.schools.ids.include?(record.organisation.id) ||
+      user.providers.ids.include?(record.organisation.id)
```

Comment:
I think this could be simplified to:

```suggestion
    user.support_user? ||
      user.user_memberships.exists?(organisation_id: record.organisation_id)
```

Rather than trying schools and then falling back to providers, it directly queries the current user's memberships as a whole. That should cut out an unnecessary database query in the case of destroying a provider.

Likewise using `record.organisation_id` in favour of `record.organisation.id` could potentially cut out another query to the database.

### For bonus points: check the organisation type, too

Technically we should really also be checking against the `organisation_type`.

```suggestion
    user.support_user? ||
      user.user_memberships.exists?(
        organisation_id: record.organisation_id,
        organisation_type: record.organisation_type,
      )
```

If our database used numeric IDs, we'd inevitably end up with users in School `1234` being able to remove users from Provider `1234` because the IDs match.

Thankfully that's not a problem for us since we're using UUIDs, so the organisation IDs will in theory always be unique. But it wouldn't hurt to tighten it up anyway – just in case.